server.go

```diff
@@ -74,12 +85,11 @@ type Server struct {
 }
 
 // NewFromConfig creates a new veneur server from a configuration specification.
-func NewFromConfig(conf Config) (ret Server, err error) {
+func NewFromConfig(conf Config, transport http.RoundTripper) (ret Server, err error) {
```

Comment:
To be honest, this entire function really needs to be refactored at this point - it's doing way too much. We could probably split it up into a `NewFromConfig` function that just initializes state (but doesn't start any goroutines) and then one or more other functions for kicking off the reads.


There may be a better approach as well, but it's not coming to me right now - let me look again in the morning!